crypto/merklekeystore/merklekeystore_test.go

```diff
@@ -0,0 +1,199 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package merklekeystore
+
+import (
+	"crypto/rand"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/stretchr/testify/require"
+	"math"
+	"testing"
+)
+
+func TestSignerCreation(t *testing.T) {
+	a := require.New(t)
+
+	h := genHashableForTest()
+	for i := uint64(0); i < 20; i++ {
+		signer, err := New(basics.Round(i), basics.Round(i+1), crypto.PlaceHolderType)
+		a.NoError(err)
+		_, err = signer.Sign(h, basics.Round(i))
+		a.NoError(err)
+	}
+
+	_, err := New(1, 0, crypto.PlaceHolderType)
+	a.Error(err)
+
+	signer, err := New(0, 0, crypto.PlaceHolderType)
+	a.NoError(err)
+	sig, err := signer.Sign(genHashableForTest(), 0)
+	a.NoError(err)
+	a.NoError(signer.GetVerifier().Verify(genHashableForTest(), sig))
+
+}
+func TestDisposableKeyPositions(t *testing.T) {
+	a := require.New(t)
+	signer, err := New(0, 100, crypto.PlaceHolderType)
+	a.NoError(err)
+
+	for i := 0; i < 100; i++ {
+		pos, err := signer.getKeyPosition(basics.Round(i))
+		a.NoError(err, i)
+		a.Equal(uint64(i), pos)
+	}
+
+	_, err = signer.getKeyPosition(100)
```

Comment:
should we move it into the loop?

Reply:
thanks for catching it, that's a bug. i'll fix it